File: vacancy-keeper/src/main/java/ru/otus/vacancykeeper/domain/Vacancy.java
package ru.otus.vacancykeeper.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(schema = "vacancy", name = "vacancy")
public class Vacancy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "scode", nullable = false)
    private String scode;

    @Column(name = "sid", nullable = false)
    private String sid;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "salary_min")
    private Long salaryMin;

    @Column(name = "salary_max")
    private Long salaryMax;

    @Column(name = "currency")
    private String currency;

    @Column(name = "schedule")
    private String schedule;

    @Column(name = "address")
    private String address;

    @Column(name = "addr_lat")
    private Float addrLat;

    @Column(name = "addr_lng")
    private Float addrLng;

    @Column(name = "employer_name")
    private String employerName;

    @Column(name = "employer_url")
    private String employerUrl;

    @Column(name = "requirement")
    private String requirement;

    @Column(name = "responsibility")
    private String responsibility;

    @Column(name = "source_url")
    private String sourceUrl;

}

File: avacancy/src/app/model/Vacancy.ts
export class Vacancy {
    id: number;
    scode: string;
    sid: string;
    name: string;
    salaryMin?: number;
    salaryMax?: number;
    currency?: string;
    schedule?: string;
    address?: string;
    addrLat?: number;
    addrLng?: number;
    employerName?: string;
    employerUrl?: string;
    requirement?: string;
    responsibility?: string;
    sourceUrl?: string;


    constructor(id: number, scode: string, sid: string, name: string, salaryMin?: number, salaryMax?: number, currency?: string, schedule?: string, address?: string, addrLat?: number, addrLng?: number, employerName?: string, employerUrl?: string, requirement?: string, responsibility?: string, sourceUrl?: string) {
        this.id = id;
        this.scode = scode;
        this.sid = sid;
        this.name = name;
        this.salaryMin = salaryMin;
        this.salaryMax = salaryMax;
        this.currency = currency;
        this.schedule = schedule;
        this.address = address;
        this.addrLat = addrLat;
        this.addrLng = addrLng;
        this.employerName = employerName;
        this.employerUrl = employerUrl;
        this.requirement = requirement;
        this.responsibility = responsibility;
        this.sourceUrl = sourceUrl;
    }
}
